packages/server/src/admin/invite.ts

```diff
@@ -193,6 +190,63 @@ async function searchForExistingProfile(request: ServerInviteRequest): Promise<P
   return undefined;
 }
 
+async function createOrUpdateProjectMembership(
+  user: User,
+  project: Project,
+  profile: ProfileResource,
+  membershipTemplate: Partial<ProjectMembership>,
+  upsert: boolean
+): Promise<ProjectMembership> {
+  const existingMembership = await searchForExistingMembership(user, project, profile);
+  if (existingMembership) {
+    if (!upsert) {
+      throw new OperationOutcomeError(badRequest('User is already a member of this project'));
+    }
+
+    // Update the existing membership
+    // Be careful to preserve the critical properties: id, project, user, and profile
+    return systemRepo.updateResource<ProjectMembership>({
+      ...existingMembership,
+      ...membershipTemplate,
+      resourceType: 'ProjectMembership',
+      id: existingMembership.id,
+      project: createReference(project),
+      user: createReference(user),
+      profile: createReference(profile),
+    });
+  }
+
+  // Otherwise, create the new membership
+  return createProjectMembership(user, project, profile, membershipTemplate);
+}
+
+async function searchForExistingMembership(
+  user: User,
+  project: Project,
+  profile: ProfileResource
+): Promise<ProjectMembership | undefined> {
+  return systemRepo.searchOne<ProjectMembership>({
+    resourceType: 'ProjectMembership',
+    filters: [
+      {
+        code: 'user',
+        operator: Operator.EQUALS,
+        value: getReferenceString(user),
+      },
+      {
+        code: 'project',
+        operator: Operator.EQUALS,
+        value: getReferenceString(project),
+      },
+      {
+        code: 'profile',
+        operator: Operator.EQUALS,
+        value: getReferenceString(profile),
```

Comment:
I'm wondering if we should even allow the same `User` to have two different memberships into the same project. 

The main use case I can think of would be if someone is both a `Practitioner` and a `Patient`, but the user for the `Patient` would be project-scoped.

I'd be in favor of removing this check, and only allowing one (User, ProjectMemership) per Project